### PR TITLE
manual: automatically generate modules documentation

### PR DESCRIPTION
**Pull request by ericsagnes**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This automatically create manual entries for modules with `meta.doc` declared.

Extra changes:
- addition of `meta.doc` for gitlab and postgresql modules (and indentation tuning)
- fix of taskserver module manual path

cc @edolstra

### CODE REVIEW

**Comment by domenkozar** [comment]
You're my hero - I'll review this shortly.

File: nixos/modules/services/misc/gitlab.nix [hunk 1]
```diff
@@ -273,179 +273,183 @@ in {
...
-      wantedBy = [ "multi-user.target" ];
-      environment = gitlabEnv;
-      path = with pkgs; [
-        config.services.postgresql.package
-        gitAndTools.git
-        ruby
-        openssh
-        nodejs
+  config = mkMerge [
+    (mkIf cfg.enable {
```
Comment (nbp) [inline nixos/modules/services/misc/gitlab.nix]: @edolstra Time to bring back `mkAlways`?

**Comment by nbp** [comment]
The patch looks good to me, except for the addition of the `mkMerge`, but I think this is an issue of the module system where we could give some extra meaning to a top-level `meta` attribute.

**Comment by ericsagnes** [comment]
@nbp +1 for a top `meta` attribute, it would make things cleaner and there doesn't seem to be a real need to have `meta` in `config` anyway.

**Comment by ericsagnes** [comment]
I made an implementation of the top level attribute approach [here](https://github.com/ericsagnes/nixpkgs/tree/modules/meta-manual).
If everybody think it is better, I can update this PR code with the top level attribute one.

**Comment by nbp** [comment]
@ericsagnes 
> […] there doesn't seem to be a real need to have meta in config anyway.

That I agree, but I do not want to add special attributes either.  special attributes tend to diverge and make the integration of modules as sub-modules harder, in general.

So my suggestion was only to have a trick in `unifyModuleSyntax` to forward the `meta` top-level attribute as a `config` option definition.  In which case we should probably also move the option declaration as part of the module system.  (hey who know, maybe one days we would have a full set of documentation for apache extensions too)

Thus, instead of doing a mkMerge in each module, we do it as part of the `unifyModuleSyntax`.

**Comment by ericsagnes** [comment]
@nbp Sorry for the late reply, I just noticed your comment today.
I added a commit that move the top level `meta` attribute into `config` in `unifyModuleSyntax`.

**Comment by domenkozar** [comment]
@ericbmerritt there are conflits for the merge.

@nbp otherwise all good?

**Comment by ericbmerritt** [comment]
@ericsagnes I suspect @domenkozar was pinging you not me.

**Comment by ericsagnes** [comment]
I noticed it didn't really work like expected while rebasing.
I will put back the WIP in the title and work a little more on this.

**Comment by ericsagnes** [comment]
I fixed the conflicts and it should be working as expected now.
@nbp could you review the code, especially the changes to `lib/modules.nix`?

**Comment by ericsagnes** [comment]
update: rebased on master to fix conflicts

**Comment by globin** [comment]
Any blockers here? @nbp @domenkozar?

**Comment by ericsagnes** [comment]
update: 
- rebased on master due to a conflict with nixos-manual module
- fixed a syntax error in wpa_supplicant module that was breaking manual generation

For the record, a side effect of this PR is to facilitate the ownership declaration of modules by moving `meta.maintainers` at top level. (no need for `mkMerge` anymore)

**Comment by nbp** [comment]
@ericsagnes Nothing more to say besides **Thanks You** !

**Comment by domenkozar** [comment]
I'm +1 for merging this. Let's wait for @edolstra and merge.

**Comment by ericsagnes** [comment]
@nbp Thanks for reviewing!
I fixed the recent conflicts with master, so this should be ready to merge.

**Comment by domenkozar** [comment]
No feedback from @edolstra, since this is very minor in terms of code but big improvements for NixOS modules declarative configuration, I'm going to merge it.

**Pull request merged**